basil/utils/USBBinds.py

```diff
@@ -73,18 +73,28 @@ def find_usb_binds(rm, log,
     results = {}
 
     for instrument in instruments:
-        # try first ASRL port
-        if instrument.get("port") and 'ASRL' not in instrument.get("port"):
-            instrument["port"] = f'ASRL{instrument["port"]}::INSTR'
-        resources = (instrument["port"],) + rm.list_resources() if instrument.get("port") else rm.list_resources()
+        log.info(f"Searching for {instrument['identification']}")
+
+        resources = rm.list_resources()
+
+        if instrument.get("port") in resources:    
```

Comment:
Since we only access the port in the instrument list to optimize the search process, the port information is not mandatory. I have now changed the code so that the port information is not accessed if this key is missing in the dictionary.